### DISCUSSION TITLE
Keycloak release plans: Text-based login flows will be removed

### DISCUSSION THREAD

**Discussion by jbman** [keycloak/keycloak#10909] [Q&A] [answered]
Having a timeline for this year is great: https://www.keycloak.org/2022/03/releases
Could someone please point out what is meant with "Text-based login flows and authenticators will be removed"?
And maybe the title should be "Keycloak release plans for 2022/2023"?

ping @stianst

**Comment by stianst** [comment, marked as answer]
If you don't know, you don't need to know ;)

Text-based login flows was implemented as POC mostly where basically built-in authenticators from KC provides a "terminal" like authentication, which was used by `kcinit`. With things like WebAuthN and IdP brokering it's useless though, and the real solution here is authz flow when there's a browser, or device flow when there isn't.

**Comment by stianst** [comment]
The post doesn't really cover our plans for 2023, and the version there is really only to include some bits that'll be removed. I'll do another post towards the end of the year for 2023 ;)
